--- openldap-2.4.6/contrib/ldapc++/src/Makefile.am~	2005-10-05 17:30:05.000000000 +0200
+++ openldap-2.4.6/contrib/ldapc++/src/Makefile.am	2007-10-31 18:51:32.000000000 +0100
@@ -44,7 +44,9 @@
 			SaslInteractionHandler.cpp \
 			StringList.cpp 
 
-include_HEADERS = LDAPAsynConnection.h \
+ldapcppincludedir = $(includedir)/ldapc++
+
+ldapcppinclude_HEADERS = LDAPAsynConnection.h \
 			LDAPAttribute.h \
 			LDAPAttributeList.h \
 			LDAPAttrType.h \
--- openldap-2.4.16/contrib/ldapc++/src/SaslInteractionHandler.cpp~	2008-04-15 01:09:26.000000000 +0200
+++ openldap-2.4.16/contrib/ldapc++/src/SaslInteractionHandler.cpp	2009-05-25 16:47:09.488348393 +0200
@@ -7,6 +7,7 @@
 #include <iostream>
 #include <iomanip>
 #include <limits>
+#include <cstdio>
 #include "config.h"
 
 #ifdef HAVE_TERMIOS_H
--- openldap-2.4.22/contrib/ldapc++/examples/startTls.cpp.orig	2010-04-15 01:50:43.000000000 +0200
+++ openldap-2.4.22/contrib/ldapc++/examples/startTls.cpp	2010-05-19 07:22:39.855313030 +0200
@@ -30,6 +30,7 @@
     
     try {
         // 1. connect using global options
+        {
         LDAPAsynConnection l(uri);
         try {
             l.start_tls();
@@ -38,9 +39,11 @@
         } catch ( LDAPException e ) {
             std::cerr << e << std::endl;
         }
+	}
 
         // 2. connect using connection specific option
-        l = LDAPAsynConnection(uri);
+	{
+        LDAPAsynConnection l(uri);
         tls=l.getTlsOptions();
         std::cout << "Current connection specific settings:" << std::endl;
         std::cout << "    CaCertfile: " << tls.getStringOption( TlsOptions::CACERTFILE) << std::endl;
@@ -58,6 +61,7 @@
         } catch ( LDAPException e ) {
             std::cerr << e << std::endl;
         }
+	}
         
         // 3. and once again using the globals
         try {
