diff -durN openldap-2.0.23.orig/include/.tb.Makefile.in.tags openldap-2.0.23/include/.tb.Makefile.in.tags
--- openldap-2.0.23.orig/include/.tb.Makefile.in.tags	Thu Jan  1 01:00:00 1970
+++ openldap-2.0.23/include/.tb.Makefile.in.tags	Tue Apr  2 14:54:40 2002
@@ -0,0 +1,6 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
+!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted/
+!_TAG_PROGRAM_AUTHOR	Darren Hiebert	/dhiebert@users.sourceforge.net/
+!_TAG_PROGRAM_NAME	Exuberant Ctags	//
+!_TAG_PROGRAM_URL	http://ctags.sourceforge.net	/official site/
+!_TAG_PROGRAM_VERSION	5.2.1	//
diff -durN openldap-2.0.23.orig/include/Makefile.in openldap-2.0.23/include/Makefile.in
--- openldap-2.0.23.orig/include/Makefile.in	Fri Jan  4 21:38:14 2002
+++ openldap-2.0.23/include/Makefile.in	Tue Apr  2 14:55:03 2002
@@ -34,7 +34,7 @@
 			-e 's;%BINDIR%;$(bindir);' \
 			-e 's;%SBINDIR%;$(sbindir);' \
 			-e 's;%LIBEXECDIR%;$(libexecdir);' \
-			-e 's;%RUNDIR%;$(localstatedir);' \
+			-e 's;%RUNDIR%;/var/run/;' \
 			-e 's;%EDITOR%;$(EDITOR);' \
 			-e 's;%FINGER%;$(FINGER);' \
 			-e 's;%SENDMAIL%;$(SENDMAIL);' \
