diff -urNp -x '*.orig' openldap-2.4.49.org/servers/slapd/Makefile.in openldap-2.4.49/servers/slapd/Makefile.in
--- openldap-2.4.49.org/servers/slapd/Makefile.in	2021-05-07 23:04:52.388862462 +0200
+++ openldap-2.4.49/servers/slapd/Makefile.in	2021-05-07 23:04:52.822195877 +0200
@@ -263,7 +263,7 @@ slapi/libslapi.la: FORCE
 	(cd slapi; $(MAKE) $(MFLAGS) all)
 
 slapd: $(SLAPD_DEPENDS) @LIBSLAPI@
-	$(LTLINK) -o $@ $(SLAPD_OBJECTS) $(LIBS) \
+	$(LTLINK) -pie -Wl,-z,defs -o $@ $(SLAPD_OBJECTS) $(LIBS) \
 		$(WRAP_LIBS)
 	$(RM) $(SLAPTOOLS)
 	for i in $(SLAPTOOLS); do \
