--- openldap-2.6.10/libraries/libldap/tls_o.c~	2025-05-22 19:56:21.000000000 +0200
+++ openldap-2.6.10/libraries/libldap/tls_o.c	2025-08-01 19:20:17.006534542 +0200
@@ -485,7 +485,8 @@ tlso_ctx_init( struct ldapoptions *lo, s
 			}
 			/* Start with the first dir in path */
 			if ( !SSL_CTX_load_verify_locations( ctx,
-				lt->lt_cacertfile, dirs[0] ) )
+				lt->lt_cacertfile, dirs[0] ) ||
+                                !SSL_CTX_set_default_verify_paths( ctx ) )
 			{
 				Debug2( LDAP_DEBUG_ANY, "TLS: "
 					"could not load verify locations (file:`%s',dir:`%s').\n",
