diff -durN ldap.orig/servers/slapd/slapd.conf ldap/servers/slapd/slapd.conf
--- ldap.orig/servers/slapd/slapd.conf	Wed Jun  9 23:42:51 1999
+++ ldap/servers/slapd/slapd.conf	Fri Oct 22 16:40:59 1999
@@ -2,13 +2,15 @@
 # See slapd.conf(5) for details on configuration options.
 # This file should NOT be world readable.
 #
-include		%SYSCONFDIR%/slapd.at.conf
-include		%SYSCONFDIR%/slapd.oc.conf
+include		/usr/share/openldap/slapd.at.conf
+include		/usr/share/openldap/slapd.at-rfc2307.conf
+include		/usr/share/openldap/slapd.oc.conf
+include		/usr/share/openldap/slapd.oc-rfc2307.conf
 schemacheck	off
 #referral	ldap://ldap.itd.umich.edu
 
-pidfile		%LOCALSTATEDIR%/slapd.pid
-argsfile	%LOCALSTATEDIR%/slapd.args
+pidfile		%LOCALSTATEDIR%/run/slapd.pid
+argsfile	%LOCALSTATEDIR%/run/slapd.args
 
 #######################################################################
 # ldbm database definitions
@@ -17,10 +19,11 @@
 database	ldbm
 suffix		"dc=your-domain, dc=com"
 #suffix		"o=Your Organization Name, c=US"
-directory	/usr/tmp
+directory	%LOCALSTATEDIR%/state/openldap
 rootdn		"cn=root, dc=your-domain, dc=com"
 #rootdn		"cn=root, o=Your Organization Name, c=US"
-rootpw		secret
+# Set your LDAP root password here:
+rootpw		{crypt}x
 # cleartext passwords, especially for the rootdn, should
 # be avoid.  See slapd.conf(5) for details.
 
