diff -Nru openldap-2.0.0/build/man.mk openldap-2.0.0.new/build/man.mk
--- openldap-2.0.0/build/man.mk	Tue Sep  5 04:34:34 2000
+++ openldap-2.0.0.new/build/man.mk	Tue Sep  5 04:36:28 2000
@@ -36,7 +36,7 @@
 			for link in `$(CAT) $(srcdir)/$$page.links`; do \
 				echo "installing $(MANDIR)/$$link as link to $$page"; \
 				$(RM) $(DESTDIR)$(MANDIR)/$$link ; \
-				$(LN_S) $$page $(DESTDIR)$(MANDIR)/$$link; \
+				echo ".so $$page" > $(DESTDIR)$(MANDIR)/$$link; \
 			done; \
 		fi; \
 	done
