diff -durN openldap-2.0.1.orig/include/ldap_defaults.h openldap-2.0.1/include/ldap_defaults.h
--- openldap-2.0.1.orig/include/ldap_defaults.h	Tue Jul  4 19:58:48 2000
+++ openldap-2.0.1/include/ldap_defaults.h	Sat Sep  9 13:11:43 2000
@@ -31,7 +31,7 @@
 
 #include <ldap_config.h>
 
-#define LDAP_CONF_FILE	 LDAP_SYSCONFDIR LDAP_DIRSEP "ldap.conf"
+#define LDAP_CONF_FILE "/etc/ldap.conf"
 #define LDAP_USERRC_FILE "ldaprc"
 #define LDAP_ENV_PREFIX "LDAP"
 
