diff -ruNp openldap-2.4.45.orig/libraries/libldap/tls_o.c openldap-2.4.45/libraries/libldap/tls_o.c
--- openldap-2.4.45.orig/libraries/libldap/tls_o.c	2017-06-01 22:01:07.000000000 +0200
+++ openldap-2.4.45/libraries/libldap/tls_o.c	2017-09-03 20:23:39.222111712 +0200
@@ -275,7 +275,8 @@ tlso_ctx_init( struct ldapoptions *lo, s
 		}
 	} else {
 		if ( !SSL_CTX_load_verify_locations( ctx,
-				lt->lt_cacertfile, lt->lt_cacertdir ) )
+				lt->lt_cacertfile, lt->lt_cacertdir ) ||
+			!SSL_CTX_set_default_verify_paths( ctx ) )
 		{
 			Debug( LDAP_DEBUG_ANY, "TLS: "
 				"could not load verify locations (file:`%s',dir:`%s').\n",
