diff -durN openldap-2.0.23.orig/include/Makefile.in openldap-2.0.23/include/Makefile.in
--- openldap-2.0.23.orig/include/Makefile.in	Fri Jan  4 21:38:14 2002
+++ openldap-2.0.23/include/Makefile.in	Tue Apr  2 14:55:03 2002
@@ -34,7 +34,7 @@
 			-e 's;%BINDIR%;$(bindir);' \
 			-e 's;%SBINDIR%;$(sbindir);' \
 			-e 's;%LIBEXECDIR%;$(libexecdir);' \
-			-e 's;%RUNDIR%;$(localstatedir);' \
+			-e 's;%RUNDIR%;/var/run/;' \
 			-e 's;%EDITOR%;$(EDITOR);' \
 			-e 's;%FINGER%;$(FINGER);' \
 			-e 's;%SENDMAIL%;$(SENDMAIL);' \
