--- openldap-2.2.8/libraries/liblber/Makefile.in.orig	Thu Jan  1 19:16:29 2004
+++ openldap-2.2.8/libraries/liblber/Makefile.in	Fri Apr  2 12:06:22 2004
@@ -47,6 +47,6 @@
 
 install-local: FORCE
 	-$(MKDIR) $(DESTDIR)$(libdir)
-	$(LTINSTALL) $(INSTALLFLAGS) -m 644 $(LIBRARY) $(DESTDIR)$(libdir)
+	$(LTINSTALL) $(INSTALLFLAGS) $(LIBRARY) $(DESTDIR)$(libdir)
 	$(LTFINISH) $(DESTDIR)$(libdir)
 
--- openldap-2.2.8/libraries/libldap/Makefile.in.orig	Thu Jan  1 19:16:29 2004
+++ openldap-2.2.8/libraries/libldap/Makefile.in	Fri Apr  2 12:06:38 2004
@@ -59,7 +59,7 @@
 
 install-local: $(CFFILES) FORCE
 	-$(MKDIR) $(DESTDIR)$(libdir)
-	$(LTINSTALL) $(INSTALLFLAGS) -m 644 $(LIBRARY) $(DESTDIR)$(libdir)
+	$(LTINSTALL) $(INSTALLFLAGS) $(LIBRARY) $(DESTDIR)$(libdir)
 	$(LTFINISH) $(DESTDIR)$(libdir)
 	-$(MKDIR) $(DESTDIR)$(sysconfdir)
 	@for i in $(CFFILES); do \
--- openldap-2.2.8/libraries/libldap_r/Makefile.in.orig	Thu Jan  1 19:16:30 2004
+++ openldap-2.2.8/libraries/libldap_r/Makefile.in	Fri Apr  2 12:06:50 2004
@@ -77,6 +77,6 @@
 
 install-local: $(CFFILES) FORCE
 	-$(MKDIR) $(DESTDIR)$(libdir)
-	$(LTINSTALL) $(INSTALLFLAGS) -m 644 $(LIBRARY) $(DESTDIR)$(libdir)
+	$(LTINSTALL) $(INSTALLFLAGS) $(LIBRARY) $(DESTDIR)$(libdir)
 	$(LTFINISH) $(DESTDIR)$(libdir)
 
