--- openldap-2.3.4/servers/slapd/Makefile.in.orig	2005-05-06 18:42:50.000000000 +0200
+++ openldap-2.3.4/servers/slapd/Makefile.in	2005-07-19 21:56:52.188705528 +0200
@@ -443,6 +443,10 @@
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	for i in $(SLAPTOOLS); do \
 		$(RM) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		if test "$(libexecdir)" = "$(sbindir)" ; then \
+			$(LN_S) slapd$(EXEEXT) $$i$(EXEEXT) ; \
+		else \
 		$(LN_S) -f $(DESTDIR)$(libexecdir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		fi ; \
 	done
 
