diff -durN openldap-2.1.12.orig/servers/slapd/back-bdb/Makefile.in openldap-2.1.12/servers/slapd/back-bdb/Makefile.in
--- openldap-2.1.12.orig/servers/slapd/back-bdb/Makefile.in	Tue Oct 29 02:24:18 2002
+++ openldap-2.1.12/servers/slapd/back-bdb/Makefile.in	Wed Jan 15 13:22:27 2003
@@ -23,7 +23,7 @@
 
 shared_LDAP_LIBS = $(LDAP_LIBLDAP_R_LA) $(LDAP_LIBLBER_LA)
 NT_LINK_LIBS = -L.. -lslapd $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
-UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
+UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS) $(LDBM_LIBS)
 
 LIBBASE = back_bdb
 
diff -durN openldap-2.1.12.orig/servers/slapd/back-perl/Makefile.in openldap-2.1.12/servers/slapd/back-perl/Makefile.in
--- openldap-2.1.12.orig/servers/slapd/back-perl/Makefile.in	Fri Jun 21 03:15:09 2002
+++ openldap-2.1.12/servers/slapd/back-perl/Makefile.in	Wed Jan 15 13:23:26 2003
@@ -28,7 +28,7 @@
 
 shared_LDAP_LIBS = $(LDAP_LIBLDAP_R_LA) $(LDAP_LIBLBER_LA)
 NT_LINK_LIBS = -L.. -lslapd $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
-UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
+UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS) @MOD_PERL_LDFLAGS@
 
 LIBBASE = back_perl
 
diff -durN openldap-2.1.12.orig/servers/slapd/back-sql/Makefile.in openldap-2.1.12/servers/slapd/back-sql/Makefile.in
--- openldap-2.1.12.orig/servers/slapd/back-sql/Makefile.in	Fri Jun 21 03:15:10 2002
+++ openldap-2.1.12/servers/slapd/back-sql/Makefile.in	Wed Jan 15 13:23:54 2003
@@ -17,7 +17,7 @@
 
 shared_LDAP_LIBS = $(LDAP_LIBLDAP_R_LA) $(LDAP_LIBLBER_LA)
 NT_LINK_LIBS = -L.. -lslapd $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
-UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
+UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS) $(SLAPD_SQL_LIBS)
 
 LIBBASE = back_sql
 
