--- openldap-2.4.22/build/openldap.m4.orig	2010-05-18 21:41:37.078938422 +0200
+++ openldap-2.4.22/build/openldap.m4	2010-05-18 21:47:47.506932556 +0200
@@ -357,7 +357,11 @@
 AC_DEFUN([OL_BERKELEY_DB_LINK],
 [ol_cv_lib_db=no
 
-if test $ol_cv_bdb_major = 4 ; then
+if test $ol_cv_bdb_major = 5 ; then
+	OL_BERKELEY_DB_TRY(ol_cv_db_db_5_dot_m,[-ldb-5.$ol_cv_bdb_minor])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db5,[-ldb5])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db,[-ldb])
+elif test $ol_cv_bdb_major = 4 ; then
 	OL_BERKELEY_DB_TRY(ol_cv_db_db_4_dot_m,[-ldb-4.$ol_cv_bdb_minor])
 	OL_BERKELEY_DB_TRY(ol_cv_db_db4m,[-ldb4$ol_cv_bdb_minor])
 	OL_BERKELEY_DB_TRY(ol_cv_db_db_4m,[-ldb-4$ol_cv_bdb_minor])
