--- openldap-2.3.20/servers/slapd/Makefile.in~	2006-02-28 12:21:11.000000000 +0200
+++ openldap-2.3.20/servers/slapd/Makefile.in	2006-02-28 12:56:13.000000000 +0200
@@ -370,7 +370,7 @@
 	done
 
 install-local-srv: install-slapd install-tools \
-	install-conf install-db-config install-schema install-tools
+	install-conf install-db-config install-schema
 
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
@@ -443,6 +443,10 @@
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	for i in $(SLAPTOOLS); do \
 		$(RM) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		if test "$(libexecdir)" = "$(sbindir)" ; then \
+			$(LN_S) $(DESTDIR)$(sbindir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		else \
 		$(LN_S) -f $(DESTDIR)$(libexecdir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		fi ; \
 	done
 
