diff -Nur openldap-2.3.34.orig/servers/slapd/back-bdb/Makefile.in openldap-2.3.34.backend_libs/servers/slapd/back-bdb/Makefile.in
--- openldap-2.3.34.orig/servers/slapd/back-bdb/Makefile.in	2007-01-02 21:43:59.000000000 +0000
+++ openldap-2.3.34.backend_libs/servers/slapd/back-bdb/Makefile.in	2007-05-16 20:06:52.000000000 +0000
@@ -39,7 +39,7 @@
 
 shared_LDAP_LIBS = $(LDAP_LIBLDAP_R_LA) $(LDAP_LIBLBER_LA)
 NT_LINK_LIBS = -L.. -lslapd $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
-UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
+UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS) $(LDBM_LIBS)
 
 LIBBASE = back_bdb
 
diff -Nur openldap-2.3.34.orig/servers/slapd/back-perl/Makefile.in openldap-2.3.34.backend_libs/servers/slapd/back-perl/Makefile.in
--- openldap-2.3.34.orig/servers/slapd/back-perl/Makefile.in	2007-01-02 21:44:06.000000000 +0000
+++ openldap-2.3.34.backend_libs/servers/slapd/back-perl/Makefile.in	2007-05-16 20:06:52.000000000 +0000
@@ -32,7 +32,7 @@
 
 shared_LDAP_LIBS = $(LDAP_LIBLDAP_R_LA) $(LDAP_LIBLBER_LA)
 NT_LINK_LIBS = -L.. -lslapd $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
-UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS)
+UNIX_LINK_LIBS = $(@BUILD_LIBS_DYNAMIC@_LDAP_LIBS) @MOD_PERL_LDFLAGS@
 
 LIBBASE = back_perl
 
