--- openldap-2.4.35/servers/slapd/Makefile.in~	2013-04-03 18:01:39.000000000 +0300
+++ openldap-2.4.35/servers/slapd/Makefile.in	2013-04-03 18:02:55.360948577 +0300
@@ -373,7 +373,7 @@
 install-dbc-no:
 
 install-local-srv: install-slapd install-tools \
-	install-conf install-dbc-maybe install-schema install-tools
+	install-conf install-dbc-maybe install-schema
 
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
@@ -443,6 +443,10 @@
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	for i in $(SLAPTOOLS); do \
 		$(RM) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		if test "$(libexecdir)" = "$(sbindir)" ; then \
+			$(LN_S) $(DESTDIR)$(sbindir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		else \
 		$(LN_S) -f $(DESTDIR)$(libexecdir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		fi ; \
 	done
 
