--- openldap-2.1.17/servers/slapd/back-perl/init.c.wigp	Mon Apr 14 23:51:53 2003
+++ openldap-2.1.17/servers/slapd/back-perl/init.c	Mon Apr 14 23:55:17 2003
@@ -25,7 +25,7 @@
 #include "perl_back.h"
 
 
-static void perl_back_xs_init LDAP_P((PERL_BACK_XS_INIT_PARAMS));
+XSINIT_t perl_back_xs_init LDAP_P((PERL_BACK_XS_INIT_PARAMS));
 EXT void boot_DynaLoader LDAP_P((PERL_BACK_BOOT_DYNALOADER_PARAMS));
 
 PerlInterpreter *PERL_INTERPRETER = NULL;
@@ -174,7 +174,7 @@
 }
 
 
-static void
+XSINIT_t
 perl_back_xs_init(PERL_BACK_XS_INIT_PARAMS)
 {
 	char *file = __FILE__;
--- openldap-2.1.17/servers/slapd/back-perl/perl_back.h.wigp	Tue Apr 15 00:19:36 2003
+++ openldap-2.1.17/servers/slapd/back-perl/perl_back.h	Tue Apr 15 00:22:42 2003
@@ -11,6 +11,9 @@
 #if defined(pTHX_) || (PERL_REVISION > 5 || (PERL_REVISION == 5 && PERL_VERSION >= 6))
 #define PERL_IS_5_6
 #endif
+#if ( PERL_REVISION > 5 || (PERL_REVISION == 5 && PERL_VERSION >= 8 ))
+#define PERL_IS_5_8
+#endif
 
 #define EVAL_BUF_SIZE 500
 
@@ -23,7 +26,7 @@
 # define na	PL_na
 #endif
 
-#if defined( HAVE_WIN32_ASPERL ) || defined( USE_ITHREADS )
+#if defined( HAVE_WIN32_ASPERL ) || defined( USE_ITHREADS ) || defined( PERL_IS_5_8 )
 /* pTHX is needed often now */
 # define PERL_INTERPRETER			my_perl
 # define PERL_BACK_XS_INIT_PARAMS		pTHX
