--- openldap-2.0.27/libraries/libldbm/ldbm.c.orig	Sat Jan  5 03:15:35 2002
+++ openldap-2.0.27/libraries/libldbm/ldbm.c	Sat Oct 26 22:01:28 2002
@@ -326,7 +326,11 @@
 
 	/* likely should use ber_mem* routines */
 
+#if (DB_VERSION_MAJOR > 4) || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR >= 1)
+	err = ret->open( ret, NULL, name, NULL, DB_TYPE, rw, mode);
+#else
 	err = ret->open( ret, name, NULL, DB_TYPE, rw, mode);
+#endif
 
 	if ( err != 0 ) {
 		int tmp = errno;
