--- openldap-2.4.6/contrib/ldapc++/src/Makefile.am~	2005-10-05 17:30:05.000000000 +0200
+++ openldap-2.4.6/contrib/ldapc++/src/Makefile.am	2007-10-31 18:51:32.000000000 +0100
@@ -44,7 +44,9 @@
                         LDAPUrlList.cpp \
                         StringList.cpp 
 
-include_HEADERS = LDAPAsynConnection.h \
+ldapcppincludedir = $(includedir)/ldapc++
+
+ldapcppinclude_HEADERS = LDAPAsynConnection.h \
                         LDAPAttribute.h \
                         LDAPAttributeList.h \
                         LDAPAttrType.h \
--- openldap-2.4.6/contrib/ldapc++/configure.in~	2007-10-02 04:24:57.000000000 +0200
+++ openldap-2.4.6/contrib/ldapc++/configure.in	2007-10-31 20:41:20.000000000 +0100
@@ -22,13 +22,6 @@
 dnl AM_DISABLE_SHARED
 AC_PROG_LIBTOOL
 dnl AC_PROG_MAKE_SET
-AC_ARG_ENABLE(debug,[  --enable-debug],[
-	CXXFLAGS="-g -O0 -Wall"
-    AC_DEFINE(WITH_DEBUG,[],[Define to 1 ot enable debug logging])
-	],[
-	CXXFLAGS="-O0"
-    ]
-)
 
 AC_ARG_WITH(libldap,[  --with-libldap=DIR          Path to the libldap library [/usr/local/lib]],[
 	LIBS="-L$with_libldap $LIBS "	
