--- openldap-2.4.35/servers/slapd/Makefile.in~	2013-04-03 18:01:39.000000000 +0300
+++ openldap-2.4.35/servers/slapd/Makefile.in	2013-04-03 18:02:55.360948577 +0300
@@ -370,7 +370,7 @@ veryclean-local-srv: FORCE
 	done
 
 install-local-srv: install-slapd install-tools \
-	install-conf install-schema install-tools
+	install-conf install-schema
 
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
@@ -443,6 +443,10 @@
 	-$(MKDIR) $(DESTDIR)$(sbindir)
 	for i in $(SLAPTOOLS); do \
 		$(RM) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		if test "$(libexecdir)" = "$(sbindir)" ; then \
+			$(LN_S) $(DESTDIR)$(sbindir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		else \
 		$(LN_S) -f $(DESTDIR)$(libexecdir)/slapd$(EXEEXT) $(DESTDIR)$(sbindir)/$$i$(EXEEXT); \
+		fi ; \
 	done
 
--- openldap-2.2.6/build/top.mk.orig	2004-01-01 19:16:25.000000000 +0100
+++ openldap-2.2.6/build/top.mk	2004-02-28 15:43:38.579652400 +0100
@@ -37,7 +37,7 @@
 libexecdir = @libexecdir@
 localstatedir = @localstatedir@
 mandir = @mandir@
-moduledir = @libexecdir@$(ldap_subdir)
+moduledir = @libdir@$(ldap_subdir)
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@$(ldap_subdir)
