--- openldap/servers/slapd/Makefile.in.orig	2019-11-05 20:46:53.842675681 +0100
+++ openldap/servers/slapd/Makefile.in	2019-11-06 21:15:57.588493379 +0100
@@ -269,6 +269,10 @@
 	for i in $(SLAPTOOLS); do \
 		$(LN_S) slapd$(EXEEXT) $$i$(EXEEXT); done
 
+slapd-shared.la: $(SLAPD_DEPENDS) @LIBSLAPI@
+	$(LTLINK) -module -shared -avoid-version -Wl,-z,defs -o $@ $(SLAPD_OBJECTS) $(LIBS) \
+		$(WRAP_LIBS) -rpath $(libexecdir)
+
 
 sslapd: version.o
 	$(LTLINK) -static -o $@ $(OBJS) version.o $(LIBS) $(WRAP_LIBS)
@@ -387,8 +391,10 @@
 		if test $$? != 0 ; then exit 1; fi ; \
 	    fi; \
 	done
+	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
+		slapd-shared.la $(DESTDIR)$(libexecdir)
 
-all-cffiles: slapd $(SLAPD_DYNAMIC_BACKENDS) dynamic_overlays
+all-cffiles: slapd slapd-shared.la $(SLAPD_DYNAMIC_BACKENDS) dynamic_overlays
 	@if test $(PLAT) = NT; then \
 	    sysconfdir=`cygpath -w $(sysconfdir) | \
 		$(SED) -e 's/\\\\/\\\\\\\\\\\\\\\\/g'`; \
