--- openldap-2.2.6/include/ldap_defaults.h.orig	2004-02-28 15:44:29.478914536 +0100
+++ openldap-2.2.6/include/ldap_defaults.h	2004-02-28 15:47:15.816627360 +0100
@@ -39,7 +39,7 @@
 #define LDAP_ENV_PREFIX "LDAP"
 
 /* default ldapi:// socket */
-#define LDAPI_SOCK LDAP_RUNDIR LDAP_DIRSEP "run" LDAP_DIRSEP "ldapi"
+#define LDAPI_SOCK "var" LDAP_DIRSEP "run" LDAP_DIRSEP "ldapi"
 
 /*
  * SLAPD DEFINITIONS
