diff -urNp -x '*.orig' openldap-2.4.49.org/contrib/ldapc++/examples/startTls.cpp openldap-2.4.49/contrib/ldapc++/examples/startTls.cpp
--- openldap-2.4.49.org/contrib/ldapc++/examples/startTls.cpp	2020-01-30 18:58:35.000000000 +0100
+++ openldap-2.4.49/contrib/ldapc++/examples/startTls.cpp	2021-05-07 23:04:51.668862326 +0200
@@ -30,6 +30,7 @@ int main( int argc, char* argv[]){
 
     try {
         // 1. connect using global options
+        {
         LDAPAsynConnection l(uri);
         try {
             l.start_tls();
@@ -38,8 +39,10 @@ int main( int argc, char* argv[]){
         } catch ( LDAPException e ) {
             std::cerr << e << std::endl;
         }
+	}
 
         // 2. connect using connection specific option
+	{
         LDAPAsynConnection l1(uri);
         tls=l1.getTlsOptions();
         std::cout << "Current connection specific settings:" << std::endl;
@@ -58,6 +61,7 @@ int main( int argc, char* argv[]){
         } catch ( LDAPException e ) {
             std::cerr << e << std::endl;
         }
+	}
 
         // 3. and once again using the globals
         try {
diff -urNp -x '*.orig' openldap-2.4.49.org/contrib/ldapc++/src/Makefile.am openldap-2.4.49/contrib/ldapc++/src/Makefile.am
--- openldap-2.4.49.org/contrib/ldapc++/src/Makefile.am	2020-01-30 18:58:35.000000000 +0100
+++ openldap-2.4.49/contrib/ldapc++/src/Makefile.am	2021-05-07 23:04:51.668862326 +0200
@@ -51,7 +51,9 @@ libldapcpp_la_SOURCES = LDAPAddRequest.c
 			StringList.cpp \
 			TlsOptions.cpp
 
-include_HEADERS = LDAPAsynConnection.h \
+ldapcppincludedir = $(includedir)/ldapc++
+
+ldapcppinclude_HEADERS = LDAPAsynConnection.h \
 			LDAPAttribute.h \
 			LDAPAttributeList.h \
 			LDAPAttrType.h \
diff -urNp -x '*.orig' openldap-2.4.49.org/contrib/ldapc++/src/SaslInteractionHandler.cpp openldap-2.4.49/contrib/ldapc++/src/SaslInteractionHandler.cpp
--- openldap-2.4.49.org/contrib/ldapc++/src/SaslInteractionHandler.cpp	2020-01-30 18:58:35.000000000 +0100
+++ openldap-2.4.49/contrib/ldapc++/src/SaslInteractionHandler.cpp	2021-05-07 23:04:51.668862326 +0200
@@ -7,6 +7,7 @@
 #include <iostream>
 #include <iomanip>
 #include <limits>
+#include <cstdio>
 #include "config.h"
 
 #ifdef HAVE_TERMIOS_H
