--- openldap-2.4.37/build/openldap.m4.orig	2013-11-01 20:06:42.762133271 +0100
+++ openldap-2.4.37/build/openldap.m4	2013-11-01 20:41:25.152045898 +0100
@@ -357,7 +357,14 @@
 AC_DEFUN([OL_BERKELEY_DB_LINK],
 [ol_cv_lib_db=no
 
-if test $ol_cv_bdb_major = 5 ; then
+if test $ol_cv_bdb_major = 6 ; then
+	OL_BERKELEY_DB_TRY(ol_cv_db_db_6_dot_m,[-ldb-6.$ol_cv_bdb_minor])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db6m,[-ldb6$ol_cv_bdb_minor])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db_6m,[-ldb-6$ol_cv_bdb_minor])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db_6_m,[-ldb-6-$ol_cv_bdb_minor])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db_6,[-ldb-6])
+	OL_BERKELEY_DB_TRY(ol_cv_db_db6,[-ldb6])
+elif test $ol_cv_bdb_major = 5 ; then
 	OL_BERKELEY_DB_TRY(ol_cv_db_db_5_dot_m,[-ldb-5.$ol_cv_bdb_minor])
 	OL_BERKELEY_DB_TRY(ol_cv_db_db5m,[-ldb5$ol_cv_bdb_minor])
 	OL_BERKELEY_DB_TRY(ol_cv_db_db_5m,[-ldb-5$ol_cv_bdb_minor])
