diff -durN ldap.orig/include/ldapconfig.h.edit ldap/include/ldapconfig.h.edit
--- ldap.orig/include/ldapconfig.h.edit	Fri Oct 22 16:57:43 1999
+++ ldap/include/ldapconfig.h.edit	Fri Oct 22 17:22:38 1999
@@ -64,6 +64,7 @@
 #define TEMPLATEFILE	"%SYSCONFDIR%/ldaptemplates.conf"
 #define SEARCHFILE	"%SYSCONFDIR%/ldapsearchprefs.conf"
 #define FRIENDLYFILE	"%DATADIR%/ldapfriendly"
+#define SECRETFILE	"%SYSCONFDIR%/secret"
 
 /*
  * FINGER DEFINITIONS
diff -durN ldap.orig/libraries/libldap/sbind.c ldap/libraries/libldap/sbind.c
--- ldap.orig/libraries/libldap/sbind.c	Thu Dec 31 20:32:07 1998
+++ ldap/libraries/libldap/sbind.c	Fri Oct 22 17:23:46 1999
@@ -20,6 +20,7 @@
 #include "lber.h"
 #include "ldap.h"
 #include "ldap-int.h"
+#include "ldapconfig.h"
 
 
 /*
@@ -32,10 +33,13 @@
  *	    "secret" )
  */
 
+char 		dn_buf[1024],pw_buf[1024];
+
 int
 ldap_simple_bind( LDAP *ld, char *dn, char *passwd )
 {
 	BerElement	*ber;
+	FILE 		*secretf;
 
 	/*
 	 * The bind request looks like this:
@@ -51,6 +55,19 @@
 
 	Debug( LDAP_DEBUG_TRACE, "ldap_simple_bind\n", 0, 0, 0 );
 
+	if ( dn == NULL && passwd == NULL ) {
+	/* try secret-file */	
+		secretf=fopen(SECRETFILE,"r");
+		if (secretf){
+			dn=fgets(dn_buf,1024,secretf);
+			if (dn && dn[0]) dn[strlen(dn)-1]=0;
+			passwd=fgets(pw_buf,1024,secretf);
+			if (passwd && passwd[0]) passwd[strlen(passwd)-1]=0;
+			fclose(secretf);
+		}
+	}
+	
+	
 	if ( dn == NULL )
 		dn = "";
 	if ( passwd == NULL )
