diff -urN openldap-2.0.8.org/servers/slapd/slapd.conf openldap-2.0.8/servers/slapd/slapd.conf
--- openldap-2.0.8.org/servers/slapd/slapd.conf	Fri May 11 17:21:08 2001
+++ openldap-2.0.8/servers/slapd/slapd.conf	Fri May 11 17:22:00 2001
@@ -3,19 +3,32 @@
 # See slapd.conf(5) for details on configuration options.
 # This file should NOT be world readable.
 #
-include		%SYSCONFDIR%/schema/core.schema
+include		/usr/share/openldap/schema/core.schema
+#include	/usr/share/openldap/schema/cosine.schema
+#include	/usr/share/openldap/schema/inetorgperson.schema
+#include	/usr/share/openldap/schema/krb5-kdc.schema.schema
+#include	/usr/share/openldap/schema/microsoft.schema
+#include	/usr/share/openldap/schema/microsoft.std.schema
+#include	/usr/share/openldap/schema/microsoft.ext.schema
+#include	/usr/share/openldap/schema/misc.schema
+#include	/usr/share/openldap/schema/nadf.schema
+#include	/usr/share/openldap/schema/nis.schema
+#include	/usr/share/openldap/schema/openldap.schema
+#include	/usr/share/openldap/schema/pilot.schema
+include		%SYSCONFDIR%/schema/local.schema
 
 # Define global ACLs to disable default read access.
+include 	%SYSCONFDIR%/slapd.access.conf
 
 # Do not enable referrals until AFTER you have a working directory
 # service AND an understanding of referrals.
 #referral	ldap://root.openldap.org
 
-pidfile		%LOCALSTATEDIR%/slapd.pid
-argsfile	%LOCALSTATEDIR%/slapd.args
+pidfile		%LOCALSTATEDIR%/run/slapd.pid
+argsfile	%LOCALSTATEDIR%/run/slapd.args
 
 # Load dynamic backend modules:
-# modulepath	%MODULEDIR%
+modulepath	/usr/lib/openldap
 # moduleload	back_ldap.la
 # moduleload	back_ldbm.la
 # moduleload	back_passwd.la
@@ -33,9 +46,9 @@
 # Cleartext passwords, especially for the rootdn, should
 # be avoid.  See slappasswd(8) and slapd.conf(5) for details.
 # Use of strong authentication encouraged.
-rootpw		secret
+#rootpw		secret
 # The database directory MUST exist prior to running slapd AND 
 # should only be accessible by the slapd/tools. Mode 700 recommended.
-directory	%LOCALSTATEDIR%/openldap-ldbm
+directory	%LOCALSTATEDIR%/lib/openldap-ldbm
 # Indices to maintain
 index	objectClass	eq
