--- openldap/servers/slapd/Makefile.in.orig	2019-11-05 20:46:53.842675681 +0100
+++ openldap/servers/slapd/Makefile.in	2019-11-06 21:15:57.588493379 +0100
@@ -269,6 +269,10 @@
 	for i in $(SLAPTOOLS); do \
 		$(LN_S) slapd$(EXEEXT) $$i$(EXEEXT); done
 
+slapd-shared.la: $(SLAPD_DEPENDS) @LIBSLAPI@
+	$(LTLINK) -module -shared -avoid-version -Wl,-z,defs -o $@ $(SLAPD_OBJECTS) $(LIBS) \
+		$(WRAP_LIBS) -rpath $(libexecdir)
+
 
 sslapd: version.o
 	$(LTLINK) -static -o $@ $(OBJS) version.o $(LIBS) $(WRAP_LIBS)
@@ -380,6 +384,8 @@
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
 	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
+	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
+		slapd-shared.la $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i && test -f $$i/Makefile ; then \
 		echo; echo "  cd $$i; $(MAKE) $(MFLAGS) install"; \
@@ -394,7 +394,7 @@
 	    fi; \
 	done
 
-all-cffiles: slapd $(SLAPD_DYNAMIC_BACKENDS) dynamic_overlays
+all-cffiles: slapd slapd-shared.la $(SLAPD_DYNAMIC_BACKENDS) dynamic_overlays
 	@if test $(PLAT) = NT; then \
 	    sysconfdir=`cygpath -w $(sysconfdir) | \
 		$(SED) -e 's/\\\\/\\\\\\\\\\\\\\\\/g'`; \
