diff -durN openldap-2.1.12.orig/build/top.mk openldap-2.1.12/build/top.mk
--- openldap-2.1.12.orig/build/top.mk	Thu Jun 20 22:12:23 2002
+++ openldap-2.1.12/build/top.mk	Wed Jan  8 11:57:09 2003
@@ -92,7 +92,7 @@
 LINK_LIBS = $(@PLAT@_LINK_LIBS)
 
 LTLINK   = $(LIBTOOL) --mode=link \
-	$(CC) -static $(LT_CFLAGS) $(LDFLAGS) $(LTFLAGS)
+	$(CC) $(LT_CFLAGS) $(LDFLAGS) $(LTFLAGS)
 
 LTCOMPILE_LIB = $(LIBTOOL) $(LTONLY_LIB) --mode=compile \
 	$(CC) $(LT_CFLAGS) $(LT_CPPFLAGS) $(LIB_DEFS) -c
