--- openldap-2.2.10/servers/slapd/Makefile.in.orig	2004-03-18 02:01:02.000000000 +0100
+++ openldap-2.2.10/servers/slapd/Makefile.in	2004-03-28 03:43:04.484125840 +0200
@@ -417,5 +417,9 @@
 	cd $(DESTDIR)$(sbindir); \
	rm -f $(SLAPTOOLS); \
 	for i in $(SLAPTOOLS); do \
-		$(LN_S) $(libexecdir)/slapd $$i; \
+		if test "$(libexecdir)" = "$(sbindir)" ; then \
+			$(LN_S) slapd $$i ; \
+		else \
+			$(LN_S) $(libexecdir)/slapd $$i; \
+		fi ; \
 	done
