--- openldap-2.4.17/servers/slapd/overlays/auditlog.c~	2009-01-22 01:01:12.000000000 +0100
+++ openldap-2.4.17/servers/slapd/overlays/auditlog.c	2009-09-07 14:53:04.758571193 +0200
@@ -73,7 +73,7 @@
 	Attribute *a;
 	Modifications *m;
 	struct berval *b, *who = NULL;
-	char *what, *suffix;
+	char *what, *whatm, *suffix;
 	time_t stamp;
 	int i;
 
@@ -147,15 +147,15 @@
 	  case LDAP_REQ_MODIFY:
 		for(m = op->orm_modlist; m; m = m->sml_next) {
 			switch(m->sml_op & LDAP_MOD_OP) {
-				case LDAP_MOD_ADD:	 what = "add";		break;
-				case LDAP_MOD_REPLACE:	 what = "replace";	break;
-				case LDAP_MOD_DELETE:	 what = "delete";	break;
-				case LDAP_MOD_INCREMENT: what = "increment";	break;
+				case LDAP_MOD_ADD:	 whatm = "add";		break;
+				case LDAP_MOD_REPLACE:	 whatm = "replace";	break;
+				case LDAP_MOD_DELETE:	 whatm = "delete";	break;
+				case LDAP_MOD_INCREMENT: whatm = "increment";	break;
 				default:
 					fprintf(f, "# MOD_TYPE_UNKNOWN:%02x\n", m->sml_op & LDAP_MOD_OP);
 					continue;
 			}
-			fprintf(f, "%s: %s\n", what, m->sml_desc->ad_cname.bv_val);
+			fprintf(f, "%s: %s\n", whatm, m->sml_desc->ad_cname.bv_val);
 			if((b = m->sml_values) != NULL)
 			  for(i = 0; b[i].bv_val; i++)
 				fprint_ldif(f, m->sml_desc->ad_cname.bv_val, b[i].bv_val, b[i].bv_len);
