diff -urN openldap-2.0.8.org/configure.in openldap-2.0.8/configure.in
--- openldap-2.0.8.org/configure.in	Fri May 11 17:21:05 2001
+++ openldap-2.0.8/configure.in	Fri May 11 17:25:38 2001
@@ -739,7 +739,7 @@
 AC_CHECK_FUNCS( getaddrinfo inet_ntop )
 
 ol_link_ipv6=no
-if test $ac_cv_func_getaddrinfo = no -o $ac_cv_func_inet_ntop ; then
+if test $ac_cv_func_getaddrinfo = no -o $ac_cv_func_inet_ntop = no ; then
 	if test $ol_enable_ipv6 = yes ; then
 		AC_MSG_ERROR([IPv6 support requires getaddrinfo() and inet_ntop()])
 	fi
